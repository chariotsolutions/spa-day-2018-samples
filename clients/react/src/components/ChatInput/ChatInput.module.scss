@import '../../mixins.scss';

.form {
  display: flex;
  margin: auto 20px 10px;
  min-height: 50px;

  input[type=text] {
    font-size: 16px;
    color: $color-text;
    height: 50px;
    padding: 0 10px;
    border-top-left-radius: 6px;
    border-bottom-left-radius: 6px;
    border: 1px solid $color-border;
    border-right: 1px solid transparent;
    box-sizing: border-box;
    flex-grow: 1;

    &:focus {
      outline: none;
    }
  }

  input[type=submit] {
    font-size: 16px;
    box-sizing: border-box;
    height: 50px;
    width: 100px;
    color: $color-white;
    background: $color-green;
    border: 1px solid $color-green;
    border-radius: 0 6px 6px 0;
    cursor: pointer;

    &:focus {
      outline: none;
    }

    &:disabled {
      cursor: auto;
      background: $color-disabled;
      border: 1px solid $color-disabled;
    }
  }
}

.error {
  box-shadow: 0 0 125px -15px red;
}
