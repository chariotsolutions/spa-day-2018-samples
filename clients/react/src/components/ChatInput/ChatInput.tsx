import styles from './ChatInput.module.scss';
import React, { Component } from 'react';
import cx from 'classnames';

interface Props {
    handleSubmit: (message: string) => void;
    placeholder?: string;
    submitText: string;
    error?: string;
}

interface State {
    message: string;
}

export class ChatInput extends Component<Props, State> {

    public state: State = {
        message: '',
    };

    public handleChange: (event: React.SyntheticEvent<HTMLInputElement>) => void =
        event => this.setState({ message: event.currentTarget.value });

    public submit: (event: React.SyntheticEvent) => void = (event) => {
        event.preventDefault();

        const message = this.state.message;
        this.setState({ message: '' }, () =>
            this.props.handleSubmit(message))
    };

    public render() {
        return (
            <form className={cx(styles.form, { [styles.error]: this.props.error })} onSubmit={this.submit}>
                <input
                    type='text'
                    value={this.state.message}
                    placeholder={this.props.error || this.props.placeholder}
                    onChange={this.handleChange}
                />

                <input
                    disabled={!this.state.message}
                    type='submit'
                    value={this.props.submitText}
                />
            </form>
        );
    }
}
