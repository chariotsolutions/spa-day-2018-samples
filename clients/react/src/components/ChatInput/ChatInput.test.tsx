import React from 'react';
import { shallow } from 'enzyme';
import { ChatInput } from './ChatInput';

describe('ChatInput', () => {
    it('snapshot regression', () => {
        const handleSubmit = jest.fn();
        const submitText = 'foo'

        expect(shallow(<ChatInput {...{ handleSubmit, submitText }}/>)).toMatchSnapshot()
    })

    fit('should update state on input change', () => {
        const handleSubmit = jest.fn();
        const submitText = 'foo'

        const component = shallow(<ChatInput {...{ handleSubmit, submitText }}/>)
        component.find('[type="text"]')
            .simulate('change',
                      { currentTarget: { value: 'hello' } })
        expect(component.state()).toEqual({ message: 'hello' })
    })

})
