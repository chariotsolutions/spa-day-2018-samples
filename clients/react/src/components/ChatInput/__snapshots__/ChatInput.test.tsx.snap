// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatInput snapshot regression 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChatInput
    handleSubmit={[MockFunction]}
    submitText="foo"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          type="text"
          value=""
        />,
        <input
          disabled={true}
          type="submit"
          value="foo"
        />,
      ],
      "className": "form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": undefined,
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "disabled": true,
          "type": "submit",
          "value": "foo",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            type="text"
            value=""
          />,
          <input
            disabled={true}
            type="submit"
            value="foo"
          />,
        ],
        "className": "form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": undefined,
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "disabled": true,
            "type": "submit",
            "value": "foo",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
